vowpalwabbit/parse_args.cc

```diff
@@ -1003,6 +1006,13 @@ void parse_reductions(vw& all)
   all.reduction_stack.push_back(bs_setup);
 
   all.l = setup_base(all);
+
+  if (all.probabilities)
+  { if (!all.vm.count("oaa") && !all.vm.count("csoaa_ldf"))
+      THROW("--probabilities can only be used with --oaa=N or --csoaa_ldf=mc");
+    if (!all.vm.count("loss_function") || all.vm["loss_function"].as<string>() != "logistic" )
```

Comment:
As I wrote in #774: I can move the `--probabilities` option to `oaa.cc` and `csoaa.cc` (i.e. duplicate its definition), but probably I will still need to store it inside `all` (similarly, e.q. `csoaa_rank` aka `multilabel_prediction` is stored inside `all` although it is valid only for some reductions). The problem would be that `vw -h` would not list the `--probabilities` options (see #665).
So should I really move the option `--probabilities` to oaa.cc and csoaa.cc?